From 6c352b8a42ec423bd25ed5e657361905afc25221 Mon Sep 17 00:00:00 2001
From: Addy Ke <addy.ke@rock-chips.com>
Date: Thu, 2 Jun 2022 15:38:26 +0800
Subject: [PATCH 09/24] net: can: rockchip: fix up the canfd/can BUS_OFF_INT
 error

BUS_OFF_INT is BIT(9), so 'isr' and 'isr_int' should NOT be u8.
Also err_int should add BUS_OFF_INT.

Signed-off-by: Addy Ke <addy.ke@rock-chips.com>
Change-Id: I7462bb68ca6a710c0b72179cc5440fce7067a0e8
---
 drivers/net/can/rockchip/rockchip_canfd.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/net/can/rockchip/rockchip_canfd.c b/drivers/net/can/rockchip/rockchip_canfd.c
index a7a93648f0a1..0cc1927678ba 100644
--- a/drivers/net/can/rockchip/rockchip_canfd.c
+++ b/drivers/net/can/rockchip/rockchip_canfd.c
@@ -601,7 +601,7 @@ static int rockchip_canfd_rx(struct net_device *ndev)
 	return 1;
 }
 
-static int rockchip_canfd_err(struct net_device *ndev, u8 isr)
+static int rockchip_canfd_err(struct net_device *ndev, u16 isr)
 {
 	struct rockchip_canfd *rcan = netdev_priv(ndev);
 	struct net_device_stats *stats = &ndev->stats;
@@ -665,9 +665,9 @@ static irqreturn_t rockchip_canfd_interrupt(int irq, void *dev_id)
 	struct net_device *ndev = (struct net_device *)dev_id;
 	struct rockchip_canfd *rcan = netdev_priv(ndev);
 	struct net_device_stats *stats = &ndev->stats;
-	u8 err_int = ERR_WARN_INT | RX_BUF_OV_INT | PASSIVE_ERR_INT |
-		     TX_LOSTARB_INT | BUS_ERR_INT;
-	u8 isr;
+	u16 err_int = ERR_WARN_INT | RX_BUF_OV_INT | PASSIVE_ERR_INT |
+		     TX_LOSTARB_INT | BUS_ERR_INT | BUS_OFF_INT;
+	u16 isr;
 	u32 dlc = 0;
 
 	isr = rockchip_canfd_read(rcan, CAN_INT);
-- 
2.30.2

